src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+                createdBook = new Book(
+                        resultSet.getString("title"),
+                        resultSet.getBigDecimal("price"));
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException(throwables);
+        }
+        return createdBook;
+    }
+
+    @Override
+    public Optional<Book> findById(int id) {
+        Book findBookById = null;
+        try (Connection connection = ConnectionUtil.connectToDatabase();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(FIND_BOOK_BY_ID_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                findBookById = new Book(bookId, title, price);
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException(throwables);
+        }
+        return Optional.ofNullable(findBookById);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> foundBooks = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.connectToDatabase();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(FIND_ALL_BOOK_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                foundBooks.add(new Book(bookId, title, price));
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException(throwables);
+        }
+        return foundBooks;
+    }
+
+    @Override
+    public Book update(Book book) {
+        Book updatedBook = null;
+        try (Connection connection = ConnectionUtil.connectToDatabase();
+                PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_BOOK_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(2, book.getTitle());
+            preparedStatement.setBigDecimal(3, book.getPrice());
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                updatedBook = new Book(bookId, title, price);
```

Comment:
The `update` method should use `preparedStatement.executeUpdate()` instead of `executeQuery()` and should not return a new `Book` object. Instead, it should return the same book object or void.